webapp/channels/src/components/dialog_router/interactive_dialog_adapter.tsx

```diff
@@ -0,0 +1,358 @@
+    };
+
+    private logError = (message: string, error?: unknown): void => {
+        console.error('[InteractiveDialogAdapter]', message, error || ''); // eslint-disable-line no-console
+    };
+
+    /**
+     * Handle validation errors from conversion
+     */
+    private handleValidationErrors = (errors: ValidationError[]): {error?: string} => {
+        if (errors.length === 0) {
+            return {};
+        }
+
+        if (this.conversionContext.enhanced) {
+            const formattedErrors = errors.map((e) => e.message);
+            const errorMessage = this.props.intl.formatMessage({
+                id: 'interactive_dialog.validation_failed',
+                defaultMessage: 'Dialog validation failed: {errors}',
+            }, {
+                errors: formattedErrors.join(', '),
+            });
+            return {error: errorMessage};
+        }
+        this.logWarn('Dialog validation errors detected (non-blocking)', {
+            errorCount: errors.length,
+            errors: errors.map((e) => ({
+                field: e.field,
+                message: e.message,
+                code: e.code,
+            })),
+            note: 'These are warnings - processing will continue for backwards compatibility',
+        });
+        return {};
+    };
+
+    private convertToAppForm = (): {form?: AppForm; error?: string} => {
+        const {elements, title, introductionText, iconUrl, submitLabel} = this.props;
+
+        const {form, errors} = convertDialogToAppForm(
+            elements,
+            title,
+            introductionText,
+            iconUrl,
+            submitLabel,
+            this.conversionContext,
+        );
+
+        const {error} = this.handleValidationErrors(errors);
+        if (error) {
+            return {error};
+        }
+
+        return {form};
+    };
+
+    /**
+     * Enhanced submission adapter with comprehensive input validation and sanitization
+     */
+    private submitAdapter = async (call: AppCallRequest): Promise<DoAppCallResult<unknown>> => {
+        try {
+            // Validate and convert AppCallRequest values back to legacy format
+            const values = call.values || {};
+            const {submission: convertedValues, errors} = convertAppFormValuesToDialogSubmission(
+                values,
+                this.props.elements,
+                this.conversionContext,
+            );
+
+            // Handle validation errors if any
+            if (errors.length > 0) {
+                this.logWarn('Form submission validation errors', {
+                    errorCount: errors.length,
+                    errors,
+                });
+            }
+
+            const legacySubmission: DialogSubmission = {
+                url: this.props.url || '',
+                callback_id: this.props.callbackId || '',
+                state: this.props.state || '',
+                submission: convertedValues as {[x: string]: string},
+                user_id: '',
```

Comment:
should these fields be populated? - like what is the server outcome if they are missing?

Reply:
No...and Yes. 
No.  
The current interactive_dialog does not implement them. So, no they are not needed for backwards compatibility.
https://github.com/mattermost/mattermost/blob/master/webapp/channels/src/components/interactive_dialog/interactive_dialog.tsx#L94-L103

Yes. 
Per docs, https://developers.mattermost.com/integrate/plugins/interactive-dialogs/#dialog-submission, submission should include the user_id as well.

However, adding it here in this PR would be a larger change than I want to make at this time.
- Also I'm not sure either are necessary. TeamID retrieve via ChannelID, userID retrieve from Session.

However, we are inconsistent and I will investigate further. Created Jira ticket
https://mattermost.atlassian.net/browse/MM-64943